src/components/manage/index.js

```diff
@@ -105,36 +108,146 @@ export class Manage extends Component {
+
+          crowdsaleContract.methods.canDistributeReservedTokens().call((err, canDistributeReservedTokens) => {
+            return canDistributeReservedTokens;
+          }).then((canDistributeReservedTokens) => {
+            console.log("#canDistributeReservedTokens:", canDistributeReservedTokens);
+            this.setState({ canDistribute: canDistributeReservedTokens })
+            resolve(this.state.canDistribute)
+          })
+          .catch(() => {
+            this.setState({ canDistribute: false })
+            resolve(this.state.canDistribute)
+          })
+        })
+    })
+  }
+
+
   canFinalize = () => {
-    const { web3Store, gasPriceStore } = this.props
-    const { web3 } = web3Store
+    const { contractStore, match } = this.props
 
     return new Promise(resolve => {
-      if (!this.state.crowdsaleHasEnded) {
+      if ((!this.state.crowdsaleHasEnded) || (this.state.shouldDistribute && this.state.canDistribute)) {
         this.setState({ canFinalize: false })
         resolve(this.state.canFinalize)
 
       } else {
-        web3.eth.estimateGas({
-          from: web3Store.curAddress,
-          to: this.state.lastContract._address,
-          data: '0x4bb278f3', // finalize signature
-          gas: web3.utils.toHex(650000),
-          gasPrice: web3.utils.toHex(gasPriceStore.slow.price),
-          value: web3.utils.toHex(0)
-        })
-          .then(() => {
-            this.setState({ canFinalize: true })
-            resolve(this.state.canFinalize)
-          })
-          .catch(() => {
-            this.setState({ canFinalize: false })
-            resolve(this.state.canFinalize)
+        attachToContract(contractStore.crowdsale.abi, match.params.crowdsaleAddress)
+          .then(crowdsaleContract => { // eslint-disable-line no-loop-func
+            console.log('attach to crowdsale contract')
+
+            if (!crowdsaleContract) return Promise.reject('No contract available')
+            crowdsaleContract.methods.finalized().call((err, finalized) => {
+              return finalized;
+            }).then((finalized) => {
+              this.setState({ canFinalize: !finalized })
+              resolve(this.state.canFinalize)
+            })
+            .catch(() => {
+              this.setState({ canFinalize: false })
+              resolve(this.state.canFinalize)
+            })
           })
       }
     })
   }
 
+  distributeReservedTokens = (addressesPerBatch) => {
+    this.updateCrowdsaleStatus()
+      .then(() => {
+        const { crowdsaleStore, contractStore } = this.props
+
+        if (!crowdsaleStore.selected.distributed && this.state.canDistribute) {
+          this.showLoader()
+
+          const crowdsaleContract = this.state.lastContract
+          crowdsaleContract.methods.token().call()
+          .then(token => {
+            attachToContract(contractStore.token.abi, token)
+            .then(tokenContract => { // eslint-disable-line no-loop-func
```

Comment:
This eslint-disable is not necessary here

Reply:
Yes, fixed.